api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/QualityImprovement2020CohortQueries.java

```diff
@@ -882,6 +882,137 @@ public CohortDefinition getMQC11DEN(String indicatorFlag) {
+   *   <li>
+   *   <li>E - All transferred IN patients
+   *   <li>
+   *   <li>F - All Transferred Out patients
+   * </ul>
+   *
+   * @return CohortDefinition
+   * @params indicatorFlag A to G For inicator 11.1 to 11.7 respectively
+   */
+  public CohortDefinition getMQC12DENP2(String indicatorFlag) {
+    CompositionCohortDefinition compositionCohortDefinition = new CompositionCohortDefinition();
+
+    compositionCohortDefinition.setName(
+            "% adultos em TARV com o mínimo de 3 consultas de seguimento de adesão na FM-ficha de APSS/PP");
+
+    compositionCohortDefinition.addParameter(new Parameter("startDate", "startDate", Date.class));
+    compositionCohortDefinition.addParameter(new Parameter("endDate", "endDate", Date.class));
+    compositionCohortDefinition.addParameter(new Parameter("location", "location", Location.class));
+
+    CohortDefinition startedART = getMQC3D1(); // A
+
+    CohortDefinition b1 =
+            getPatientsFromFichaClinicaB1OrB2("B1"); // B1
+
+    CohortDefinition b1E =
+            getPatientsFromFichaClinicaB1OrB2("B1E"); // B1E
+
+    CohortDefinition b2 =
+            getPatientsFromFichaClinicaB1OrB2("B2_12"); // B2
+
+    CohortDefinition b2E =
+            getPatientsFromFichaClinicaB1OrB2("B2E"); // B2
+
+    // CohortDefinition nutritionalClass = getPatientsWithNutritionalState();
+
+    CohortDefinition pregnant =
+            commonCohortQueries.getMohMQPatientsOnCondition(
+                    true,
+                    false,
+                    "once",
+                    hivMetadata.getMasterCardEncounterType(),
+                    commonMetadata.getPregnantConcept(),
+                    Collections.singletonList(hivMetadata.getYesConcept()),
+                    null,
+                    null);
+
+    CohortDefinition breastfeeding =
+            commonCohortQueries.getMohMQPatientsOnCondition(
+                    true,
+                    false,
+                    "once",
+                    hivMetadata.getMasterCardEncounterType(),
+                    commonMetadata.getBreastfeeding(),
+                    Collections.singletonList(hivMetadata.getYesConcept()),
+                    null,
+                    null);
+
+    CohortDefinition transferIn =
+            commonCohortQueries.getMohMQPatientsOnCondition(
+                    false,
+                    true,
+                    "once",
+                    hivMetadata.getMasterCardEncounterType(),
+                    commonMetadata.getTransferFromOtherFacilityConcept(),
+                    Collections.singletonList(hivMetadata.getYesConcept()),
+                    hivMetadata.getTypeOfPatientTransferredFrom(),
+                    Collections.singletonList(hivMetadata.getArtStatus()));
+
+    CohortDefinition transfOut = commonCohortQueries.getTranferredOutPatients();
+
+    compositionCohortDefinition.addSearch("A", EptsReportUtils.map(startedART, MAPPING));
+
+    compositionCohortDefinition.addSearch(
```

Comment:
Here you need to adjust the mappings as the specs says, some how our mappings need to subtract 14 and 11 months from end date then be applied as start and end dates respectively

Reply:
well noted... Thank you !